circuit/src/verifier/transcript/mod.rs

```diff
@@ -354,46 +366,497 @@ mod test {
+			);
+
+			let mut poseidon_read =
+				PoseidonReadChipset::<_, C, _, P, R>::new(self.reader.as_slice(), loader);
+			let res = poseidon_read.read_scalar().unwrap();
+
+			let mut lb = layouter_rc.lock().unwrap();
+			lb.constrain_instance(res.inner.clone().cell(), config.common.instance, 0)?;
+
+			Ok(())
+		}
 	}
-	#[ignore = "TODO."]
+
 	#[test]
 	fn test_read_scalar() {
 		// Test read scalar
+		let rng = &mut thread_rng();
+		let random = Scalar::random(rng);
+		let mut reader = Vec::new();
+		reader.write_all(random.to_bytes().as_slice()).unwrap();
+		let mut poseidon_read =
+			PoseidonRead::<_, G1Affine, Bn256_4_68, Params>::new(reader.as_slice(), NativeSVLoader);
+
+		let res = poseidon_read.read_scalar().unwrap();
+		let circuit = TestReadScalarCircuit::new(reader);
+		let k = 7;
+		let prover = MockProver::run(k, &circuit, vec![vec![res]]).unwrap();
+		assert_eq!(prover.verify(), Ok(()));
+	}
+
+	#[derive(Clone)]
+	struct TestReadEcPointCircuit {
+		reader: Vec<u8>,
+	}
+
+	impl TestReadEcPointCircuit {
+		fn new(reader: Vec<u8>) -> Self {
+			Self { reader }
+		}
+	}
+
+	impl Circuit<Scalar> for TestReadEcPointCircuit {
+		type Config = TestConfig;
+		type FloorPlanner = SimpleFloorPlanner;
+
+		fn without_witnesses(&self) -> Self {
+			self.clone()
+		}
+
+		fn configure(meta: &mut ConstraintSystem<Scalar>) -> TestConfig {
+			let common = CommonConfig::new(meta);
+			let main_selector = MainChip::configure(&common, meta);
+			let main = MainConfig::new(main_selector);
+
+			let full_round_selector = FullRoundHasher::configure(&common, meta);
+			let partial_round_selector = PartialRoundHasher::configure(&common, meta);
+			let poseidon = PoseidonConfig::new(full_round_selector, partial_round_selector);
+
+			let absorb_selector = AbsorbChip::<Scalar, WIDTH>::configure(&common, meta);
+			let poseidon_sponge = PoseidonSpongeConfig::new(poseidon.clone(), absorb_selector);
+
+			let bits2num = Bits2NumChip::configure(&common, meta);
+
+			let int_red =
+				IntegerReduceChip::<Base, Scalar, NUM_LIMBS, NUM_BITS, P>::configure(&common, meta);
+			let int_add =
+				IntegerAddChip::<Base, Scalar, NUM_LIMBS, NUM_BITS, P>::configure(&common, meta);
+			let int_sub =
+				IntegerSubChip::<Base, Scalar, NUM_LIMBS, NUM_BITS, P>::configure(&common, meta);
+			let int_mul =
+				IntegerMulChip::<Base, Scalar, NUM_LIMBS, NUM_BITS, P>::configure(&common, meta);
+			let int_div =
+				IntegerDivChip::<Base, Scalar, NUM_LIMBS, NUM_BITS, P>::configure(&common, meta);
+
+			let ladder = EccUnreducedLadderConfig::new(int_add, int_sub, int_mul, int_div);
+			let add = EccAddConfig::new(int_red, int_sub, int_mul, int_div);
+			let double = EccDoubleConfig::new(int_red, int_add, int_sub, int_mul, int_div);
+			let table_select = EccTableSelectConfig::new(main.clone());
+			let ecc_mul_scalar = EccMulConfig::new(ladder, add, double, table_select, bits2num);
+			TestConfig { common, main, poseidon_sponge, ecc_mul_scalar }
```

Comment:
I see that this duplicate in every test circuit.
Perhaps you could add a constructor for TestCircuit that does this